tests/integration_tests/node_registry/test_node_registry.py

```diff
@@ -0,0 +1,303 @@
+import pytest
+from mipengine.node_registry.node_registry import NodeRegistryClient
```

Comment:
can you try to run the tests once more? It now starts a new containerized consul on a different port (9500) than the consul container used for the rest of the system. After all tests run the container is removed. 
PS. the 1st fixture makes a system call docker run... to start the container. Your system needs to allow execute docker commands without sudo otherwise it will fail. Also port 9500 must be free, otherwise it will fail

Reply:
After this is merged tests will run automatically on every commit.

https://github.com/madgik/MIP-Engine/pull/98